src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/MoreTimeRequestedCallbackHandler.java

```diff
@@ -0,0 +1,271 @@
+    private static final String ERROR_MESSAGE =
+        "There was a technical problem. Nothing has been sent. You need to try again.";
+    private static final String STANDARD_DEADLINE_TEXT = "You’ve been given an extra 14 days to respond to the"
+            + " claim made against you by %s.\n"
+            + "\n"
+            + "You now need to respond to the claim before 4pm on %s.\n"
+            + "\n"
+            + "If you don’t respond, you could get a County Court Judgment (CCJ). This may make it harder to get "
+            + "credit, such as a mobile phone contract, credit card or mortgage.";
+    private static final String RESPONSE_DEADLINE_PREVIEW = "responseDeadlinePreview";
+
+    @Autowired
+    public MoreTimeRequestedCallbackHandler(
+            EventProducer eventProducer,
+            ResponseDeadlineCalculator responseDeadlineCalculator,
+            MoreTimeRequestRule moreTimeRequestRule,
+            CaseDetailsConverter caseDetailsConverter,
+            NotificationService notificationService,
+            NotificationsProperties notificationsProperties,
+            GeneralLetterService generalLetterService,
+            UserService userService,
+            @Value("${doc_assembly.generalLetterTemplateId}") String generalLetterTemplateId) {
+        this.eventProducer = eventProducer;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+        this.moreTimeRequestRule = moreTimeRequestRule;
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.generalLetterService = generalLetterService;
+        this.notificationService = notificationService;
+        this.notificationsProperties = notificationsProperties;
+        this.userService = userService;
+        this.generalLetterTemplateId = generalLetterTemplateId;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_START, this::requestMoreTimeViaCaseworker,
+            CallbackType.ABOUT_TO_SUBMIT, this::sendNotifications
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    public AboutToStartOrSubmitCallbackResponse requestMoreTimeViaCaseworker(CallbackParams callbackParams) {
+        CallbackRequest callbackRequest = callbackParams.getRequest();
+        Claim claim = caseDetailsConverter.extractClaim(callbackRequest.getCaseDetails());
+        CCDCase ccdCase = caseDetailsConverter.extractCCDCase(callbackRequest.getCaseDetails());
+        LocalDate newDeadline = responseDeadlineCalculator.calculatePostponedResponseDeadline(claim.getIssuedOn());
+
+        List<String> validationResult = this.moreTimeRequestRule.validateMoreTimeCanBeRequested(claim);
+        AboutToStartOrSubmitCallbackResponseBuilder builder = AboutToStartOrSubmitCallbackResponse
+            .builder();
```

Comment:
Good candidate for `var` - it's clearly a builder for `AboutToStartOrSubmitCallbackResponse`